src/Disks/IDisk.cpp

```diff
@@ -126,4 +129,87 @@ SyncGuardPtr IDisk::getDirectorySyncGuard(const String & /* path */) const
     return nullptr;
 }
 
+void IDisk::startup(ContextPtr context, bool skip_access_check)
+{
+    if (!skip_access_check)
+    {
+        if (isReadOnly())
+        {
+            LOG_DEBUG(&Poco::Logger::get("IDisk"),
```

Comment:
The idea was to mark it readonly only if it was marked as read-only in the first place, example - web disk.

Reply:
yes, but even explicitly marked as read only disks should also be checked for read access even if this check is not really applicable for web disk?

Reply:
The thing with this access check that it first need to create a file to check if disk is readable or not, but for read-only it is not possible.

Reply:
yeah, this is why I wrote above "I understand that it is not doable in the general case, but for local disk it is, so it could be noop if not possible."

Reply:
But indeed this is not necessary, I do not force on this

Reply:
I thought about this, and for now I think it is better to firstly check access and the the brokeness/readonly check, and see how this will work out in production. If this will not suites, then change it.

Reply:
Actually for local disk, it will be work as you suggesting, since there is a check before creating directory:

```
void DiskLocal::checkAccessImpl(const String & path)
{
    try
    {
        fs::create_directories(disk_path);
    }
    catch (...)
    {
        LOG_ERROR(logger, "Cannot create the directory of disk {} ({}).", name, disk_path);
        readonly = true;
        return;
    }

    IDisk::checkAccessImpl(path);
}
```

But only when directory does not exists, I'm fine to add `canWrite` there, to keep behavior consistent.

Reply:
Done in https://github.com/ClickHouse/ClickHouse/pull/43143/commits/e6e223adc2c9aa380703f77abed15c48415e515f and https://github.com/ClickHouse/ClickHouse/pull/43143/commits/c393af2812cb56a60a99d48c15c0443d2da98a8b